### PR TITLE
Fix TSmapEstimator stat_scan with sum_over_energy_group=False 

### PR DESCRIPTION
**Pull request by QRemy**
Fix TSmapEstimator for options sum_over_energy_group=False and selection_optional=["stat_scan"]

### CODE REVIEW

File: gammapy/estimators/map/tests/test_ts.py [hunk 1]
```diff
@@ -564,12 +564,36 @@ def test_joint_ts_map_hawc():
...
 
-    estimator = TSMapEstimator(kernel_width=2 * u.deg, sum_over_energy_groups=False)
+    estimator = TSMapEstimator(
+        kernel_width=2 * u.deg, sum_over_energy_groups=False, n_jobs=4
+    )
+    result = estimator.run(datasets)
+    assert_allclose(result["flux"].data[0, 59, 59], 1.909396e-13, rtol=1e-3)
+    assert_allclose(result["sqrt_ts"].data[0, 59, 59], 10.878956, rtol=1e-3)
+
+    estimator = TSMapEstimator(
```
Comment (AtreyeeS) [inline gammapy/estimators/map/tests/test_ts.py:574]: Isnt the test getting repeated here?

File: gammapy/estimators/map/ts.py [hunk 2]
```diff
@@ -511,7 +511,7 @@ def estimate_flux_map(self, datasets):
         j, i = zip(*positions)
 
         geom = maps[0]["counts"].geom.squash(axis_name="energy")
-        energy_axis = maps[0]["counts"].geom.axes["energy"]
+        energy_axis = geom.axes["energy"]
```
Comment (AtreyeeS) [inline gammapy/estimators/map/ts.py:514]: This seems a cleanup.

**Review by AtreyeeS — COMMENTED** [review]
What was the error and what is the fix here? It seems you made one cleanup in the estimator.

File: gammapy/estimators/map/ts.py (hunk 2 above)
Comment (QRemy) [reply gammapy/estimators/map/ts.py:514]: geom is squashed but not on the energy axis line

File: gammapy/estimators/map/tests/test_ts.py (hunk 1 above)
Comment (QRemy) [reply gammapy/estimators/map/tests/test_ts.py:574]: `selection_optional=["stat_scan"]` is added independently in the second case to be sure it returns the same.
I could add a test on the scan output.

**Review by AtreyeeS — APPROVED** [review]
Thanks @QRemy !

**Pull request merged**